File: src/main/java/com/github/xumeng/java/problems/easy/problem0231/Solution0231S2.java
package com.github.xumeng.java.problems.easy.problem0231;

public class Solution0231S2 {

    public boolean isPowerOfTwo(int n) {
        if (n <= 0) {
            return false;
        }
        int countOfOne = 0;
        for (int quotient = n / 2; quotient > 0; n = quotient) {
            quotient = n / 2;
            if (n % 2 == 1) {
                countOfOne++;
            }
            if (countOfOne > 1) {
                return false;
            }
        }
        return true;
    }

}

File: src/test/kotlin/com/github/xumeng/kotlin/problems/easy/problem0231/Solution0231S2.kt
package com.github.xumeng.kotlin.problems.easy.problem0231

class Solution0231S2 {

    fun isPowerOfTwo(n: Int): Boolean {
        if (n <= 0) {
            return false
        }
        var number = n
        var countOfOne = 0
        var quotient = number / 2
        while (quotient > 0) {
            quotient = number / 2
            if (number % 2 == 1) {
                countOfOne++
            }
            if (countOfOne > 1) {
                return false
            }
            number = quotient
        }
        return true
    }

}